File: docs/vs-2015/snippets/visualbasic/VS_Snippets_VSSDK/dynamictoolboxmembers/vb/toolboxconfig.vb
'<Snippet03>
Imports Microsoft.VisualStudio.Shell
Imports System.Collections
Imports System.ComponentModel
Imports System.Drawing.Design
Imports System.Reflection
Imports System.Runtime.InteropServices
Imports System.Security.Permissions
'</Snippet03>

'<Snippet04>
<ProvideAssemblyFilter("ItemConfigurationVB, Version=*, Culture=*, PublicKeyToken=*")> _
<Guid("4DDC7895-442A-45e7-82E7-4E85F243C321")> _
Public Class ToolboxConfig
    Implements IConfigureToolboxItem
    '</Snippet04>

    '<Snippet05>
    <PrincipalPermission(SecurityAction.Demand)> Public Sub _
        ConfigureToolboxItem(ByVal item As ToolboxItem) _
        Implements IConfigureToolboxItem.ConfigureToolboxItem

        If item Is Nothing Then Return

        ' Create a filter for the Toolbox.
        Dim newFilter As ToolboxItemFilterAttribute
        If GetType(Control1).ToString() = item.TypeName Then

            ' For Control1, only show it when editing a UserControl.
            newFilter = New ToolboxItemFilterAttribute( _
                "System.Windows.Forms.UserControl", _
                ToolboxItemFilterType.Require)

        ElseIf GetType(Control2).ToString() = item.TypeName Then

            ' For Control2, only show it when not editing a UserControl.
            newFilter = New ToolboxItemFilterAttribute( _
                "System.Windows.Forms.UserControl", _
                ToolboxItemFilterType.Prevent)

        Else

            ' Don't apply a filter to other classes.
            Return

        End If

        item.Filter = CType(New ToolboxItemFilterAttribute() {newFilter}, ICollection)
    End Sub
    '</Snippet05>

End Class

File: docs/vs-2015/snippets/csharp/VS_Snippets_VSSDK/dynamictoolboxmembers/cs/toolboxconfig.cs
using System;

//<Snippet03>
using Microsoft.VisualStudio.Shell;
using System.Collections;
using System.ComponentModel;
using System.Drawing.Design;
using System.Reflection;
using System.Runtime.InteropServices;
using System.Security.Permissions;
//</Snippet03>

namespace Company.ItemConfigurationCS
{
    //<Snippet04>
    [ProvideAssemblyFilter("ItemConfigurationCS, Version=*, Culture=*, PublicKeyToken=*")]
    [Guid("E6832593-BF07-4de1-AA0F-7F9B94887DB8")]
    public class ToolboxConfig : IConfigureToolboxItem
    //</Snippet04>
    {
        //<Snippet05>
        [PrincipalPermission(SecurityAction.Demand)]
        public void ConfigureToolboxItem(ToolboxItem item)
        {
            if (null == item) return;

            // Create a filter for the Toolbox.
            ToolboxItemFilterAttribute newFilter;
            if (typeof(Control1).ToString() == item.TypeName)
            {
                // For Control1, only show it when editing a UserControl.
                newFilter = new ToolboxItemFilterAttribute(
                    "System.Windows.Forms.UserControl",
                    ToolboxItemFilterType.Require);
            }
            else if (typeof(Control2).ToString() == item.TypeName)
            {
                // For Control2, only show it when not editing a UserControl.
                newFilter = new ToolboxItemFilterAttribute(
                    "System.Windows.Forms.UserControl",
                    ToolboxItemFilterType.Prevent);
            }
            else
            {
                // Don't apply a filter to other classes.
                return;
            }

            item.Filter = (ICollection)(new ToolboxItemFilterAttribute[] { newFilter });
        }
        //</Snippet05>
    }
}
